File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.Stack;

class Solution {
    public int solution(String s) {
        int answer = 0;
        

        for(int i = 0; i < s.length(); i++) {
        	Stack<Character> st = new Stack<>();
            for(int j = 0; j < s.length(); j++) {
                if(j + i >= s.length()) {
                	st.add(s.charAt(j + i - s.length()));
                }
                else st.add(s.charAt(j + i));
                
                
                if(st.size() >= 2 && ((st.get(st.size() - 2) - st.peek() == -1) || (st.get(st.size() - 2) - st.peek() == -2))) {
                	st.pop();
                	st.pop();
                }
            }
            
            if(st.isEmpty()) answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0

    pairs = {
        "(" : ")",
        "{" : "}",
        "[" : "]"
    }

    for i in range(0, len(s)):
        rotated = list(s[i:] + s[:i])

        stack = []

        while rotated:
            current = rotated.pop(0)
            prev = stack[-1] if stack != [] else -1

            if prev in pairs.keys() and pairs[prev] == current:
                stack.pop()
            else:
                stack.append(current)    

        if stack == []:
            answer += 1        

    return answer